tests/testthat/test-compiler-errors.R

```diff
@@ -197,3 +196,33 @@ test_that("animint() is an error", {
     animint()
   }, "No arguments passed to animint. Arguments should include ggplots(1 or more) and options(0 or more)", fixed=TRUE)
 })
+
+test_that("Same argument passed to aes and geom is an error", {
+  scatter <- ggplot()+geom_path(aes(
+    x=life.expectancy, 
+    y=fertility.rate, 
+    color=region,
+    group=country, alpha=population),alpha=0.5,
+    data=WorldBank)
+  viz <- list(
+    plot = scatter
+  )
+  expect_error({
+      animint2dir(viz, open.browser=FALSE)
+  }, "Same argument cannot be passed to both aes and geom. Argument passed to both aes and geom: alpha")
+})
+
+test_that("alpha and alpha_off passed to aes and geom is an error", {
+  scatter <- ggplot()+geom_path(aes(
+    x=life.expectancy, 
+    y=fertility.rate, 
+    color=region,
+    group=country, alpha=population),alpha_off=0.5,
+    data=WorldBank)
+  viz <- list(
+    plot = scatter
+  )
+  expect_error({
+    animint2dir(viz, open.browser=FALSE)
+  }, "Same argument cannot be passed to both aes and geom. Argument passed to both aes and geom: alpha")
```

Comment:
can you please change the error message? it is not true that alpha was passed to both aes and geom, because alpha_off was passed to geom.
Also it would be good to explain why the same argument cannot be used / how to fix: the visual property needs only be defined in one place, so if the alpha should be different for each rendered geom, but not depend on selection state, then use aes(alpha); but if the alpha should depend on the selection state then use geom(alpha,alpha_off).
and actually for this test to make more sense with alpha_off, you would need to add clickSelects="some_selection_variable" so that the alpha_off could actually be used.
In fact could you please add another test for an error message when alpha_off is used in a geom without clickSelects? that should be a different error (alpha_off defines the opacity of geoms with clickSelects, but are not currently selected, so please fix by removing alpha_off or adding clickSelects)

Reply:
I updated the error message to distinguish between alpha and alpha_off. Is the new error message better to understand for the user ?

Added a new test without clickSelects and alpha_off and verified that the correct warning is displayed. When aplha_off is used without clickSelects, currently a warning is thrown. Does it make more sense to throw an error instead ?

Reply:
Currently if there are multiple errors, one error masks the others since the compiler stops upon an error. Maybe this can be modified to throw multiple errors before stopping when its possible to continue compilation. gcc/llvm do that and continue compilation to discover additional errors